pdf-viewer: Show pdf files in a new window.

akashnimare opened the pull request:
This adds a feature of showing the pdf attachments in a new window so that a user can quickly view the same.

Fixes: #547.



**Screenshots?**

<img src="https://cl.ly/ed01deb1bb01/download/Screen%20Recording%202018-08-29%20at%2006.33%20PM.gif"/>

**You have tested this PR on:**
  - [ ] Windows
  - [ ] Linux/Ubuntu
  - [x] macOS

akashnimare committed:
pdf-viewer: Show pdf files in a new window.

This adds a feature of showing the pdf attachments in a
new window so that a user can quickly view the same.

Fixes: #547.

akashnimare commented:
@abhigyank can you cross-test this?

abhigyank reviewed (commented):
The PDF opens in a new window as expected in a new window for CZO.
For organizations on zulipchat - asks me to login into the realm. I think we need to pass the session to the pdfWindow.
Clicking on download - pops up the dialog box to download to a folder. This is not connected to our download notification system hence no notification on download completion.

abhigyank commented on app/main/index.js:
```diff
@@ -196,6 +196,28 @@ app.on('ready', () => {
+			width: mainWindowState.width - paddingWidth,
+			height: mainWindowState.height - paddingHeight,
+			webPreferences: {
+				plugins: true
```
I think we need to pass the `partition: 'persist:webviewsession'` parameter to transfer the realm session into this BrowserWindow.

akashnimare replied:
Oh yes, I forgot. Thanks!

akashnimare committed:
pdf-viewer: Persist webview session.

This passes the partition parameter to the pdf window
so that a user don't need to login in order to view the
pdf.

abhigyank commented:
@akashnimare couple of things in this PR - 
* Clicking on download - pops up the dialog box to download to a folder. This is not connected to our download notification system hence no notification on download completion. Is there a way that we could capture this download and show a notification upon completion?
* In the pdf window - I don't think we should have the top menu bar (that includes options of File,View etc).

akashnimare commented:
> Is there a way that we could capture this download and show a notification upon completion?

I doubt it and it's okay to not have the notification option I think. I looked into it a bit but couldn't find any way to trigger the notification event when a user clicks on the download button.

> In the pdf window - I don't think we should have the top menu bar (that includes options of File,View etc).

There is no menu bar in the macOS. Not sure how to hide it in the Windows.

abhigyank commented:
There is a menu bar in Linux as well. I'll see if it is possible to hide it.

akashnimare commented:
> There is a menu bar in Linux as well. I'll see if it is possible to hide it.

Yeah, setting the menu to be `null` does the job here -
 > `pdfWindow.setMenu(null)`
